### PR TITLE
Datatypes update

### PR DESCRIPTION
**Pull request by pmslavin**
This relates to [HWI issue 65](https://gitlab.hydra.org.uk/hydra/hwi/issues/65) over on GitLab.

There are *definitely* some further updates and extensions required to this code relating to the other datatypes issues, but this represents a milestone that has all of the functionality of the previous solution, yet uses the new approach.

### CODE REVIEW

**Comment by pmslavin** [comment]
There are also some pytests for the new types, all of which should pass as below...

datatype tests

These are somewhat ad hoc - it may be useful to talk about standardising the format of our tests.  I'll take a look at the existing tests as a prelude to this.

**Comment by jetuk** [comment]
It would be good to get the CI working on Travis. So we don't just ignore it.

**Comment by pmslavin** [comment]
This now contains the `Timeseries` and `DataFrame` types, the `.value` property, more tests, more rigorous validation, and some reorganization of related `hydra-base` code (e.g. the replacement of `set_val()` with a direct assignment, and the replacement of `unicode` with `six` references).

There are also some comments and documentation of the interface specified by the `DataType` abc which allows `help(DataType)`.

Note that this branch requires the `hwi` and `template_manager` changes identified by the `datatypes-xx` tag attached to commits in those repos. These provide the front-end support (browser widgets etc.) for the new types. The `xx` sequence numbers identify versions which share a common interface and so work together nicely.  These will be incremented in the event that any interface-breaking changes are introduced as part of the review process.

File: hydra_base/lib/HydraTypes/Types.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    value = property(get_value, set_value)
+
+
+class Timeseries(DataType):
+    tag      = "TIMESERIES"
+    skeleton = "[%s, ...]"
+    json     = TimeseriesJSON()
+
+    def __init__(self, ts):
+        self.value = ts.to_json(date_format='iso', date_unit='ns')
```
Comment (knoxsp) [inline hydra_base/lib/HydraTypes/Types.py]: Can you make this look like the dataframe structure, whereby the to_json is set in the get_value, and the object deals with dataframes for _value and not strings.

**Comment by pmslavin** [comment]
5372989 has DataFrame test validity on set_value, and TimeSeries store `._value` as Pandas type.

**Pull request merged**